artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ConsumerListener.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server;
+
+/**
+ */
```

Comment:
I don't like this...

We have the SessionCallback to support protocol specific things.

the protocolData should be accessed by Protocol specific classes only only. We shouldn't be doing any sort of weird conversions here. 

I'm trying to keep the implementation clean and easy to understand.

Add a method on SessionCallback:

canceledDelivery(ServerConsumer, MessageReference, boolean failed);

And simply call:

callback.canceledDelivery(this, ref, failed) from ServerConsumer where you were calling ConsumerListener,


And remove this interface please.